Automattic/wp-calypso
# Ecommerce Segmentation Survey: Create Survey main component and routing provider

Closes https://github.com/Automattic/wp-calypso/issues/88920 

## Proposed Changes

* Create SurveyContainer component: this is the main component that will receive the list of questions and render using the stepper layout and calling pagination functions. It depends on a context that handles pagination, and each survey should implement its own provider.
* Create EcommerceSegmentationSurvey step: this is the Step implementation. Its folder also contains the questions file, where the questions are defined (for now just a mock survey). Under a feature flag `ecommerce-segmentation-survey`.

<img width="1040" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3113712/3988bddc-4d85-4c21-8c5a-a973c502e630">

**TODOs**
- Design details not fully covered as part of this PR
- Missing to apply on the e-commerce flow (under a feature flag)

## Testing Instructions

* Apply this PR to your local
* Revert the last commit to access the testing route `git revert cfe3816`
* Go to http://calypso.localhost:3000/ecommerce-segmentation-survey?flags=ecommerce-segmentation-survey
* You should see the mock survey
* You should be able to navigate between the questions (pressing back, skip, and continue)
* You should see the query params updated with the current page
* When loading the page with a predefined a page, the related question should be displayed

## Pre-merge Checklist
